### PR TITLE
Migrate kinesis connector to use AWS SDK v2

### PR DESCRIPTION
**Pull request by ankushChatterjee**
## Description
AWS Java SDK V2 is a rewrite of the AWS Java SDK v1. This PR migrates the kinesis connector to use AWS SDK v2.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -174,7 +174,7 @@
         <dep.iceberg.version>1.3.0</dep.iceberg.version>
         <dep.protobuf.version>3.22.2</dep.protobuf.version>
         <dep.wire.version>4.5.0</dep.wire.version>
-        <dep.netty.version>4.1.92.Final</dep.netty.version>
+        <dep.netty.version>4.1.94.Final</dep.netty.version>
```
Comment (wendigo) [inline pom.xml:177]: Do you need 94? It's not working with apache arrow 12.0.1

Comment (wendigo) [reply pom.xml:177]: We need to wait for the release than contains https://github.com/apache/arrow/issues/36209

Comment (mosabua) [reply pom.xml:177]: We are now at   <version>4.1.101.Final</version>

**Comment by mosabua** [comment]
:wave: @ankushChatterjee - this PR has become inactive. We hope you are still interested in working on it. Please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Comment by ankushChatterjee** [comment]
@mosabua I was working on this as a part of the work for my last employer. I will re-start work on this in my capacity. If this has too many merge conflicts, we should close this, I will inform on this thread.

**Comment by ankushChatterjee** [comment]
Please close this. I wont be working on this for now. If I resume work on this I will raise a new PR.

**Comment by wendigo** [comment]
Can I take over it? @ankushChatterjee

**Comment by ankushChatterjee** [comment]
@wendigo absolutely, please go ahead.

**Comment by mosabua** [comment]
Reminder @wendigo ...

**Comment by wendigo** [comment]
Continued here: https://github.com/trinodb/trino/pull/21314

**Pull request closed without merging**